test/OpenTelemetry.Tests/Logs/LoggerProviderTests.cs

```diff
@@ -0,0 +1,114 @@
+        }
+
+        [Fact]
+        public void VerifyResourceBuilderAddService()
+        {
+            InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider, configure: options => options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(serviceName: "MyService", serviceVersion: "1.2.3")));
+
+            Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.ToString() == "MyService");
+            Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceVersion && kvp.Value.ToString() == "1.2.3");
+        }
+
+        [Fact]
+        public void VerifyResourceBuilder_WithServiceNameEnVar()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(OtelServiceNameEnvVarDetector.EnvVarKey, "MyService");
+
+                InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider);
+
+                Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.ToString() == "MyService");
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(OtelServiceNameEnvVarDetector.EnvVarKey, null);
+            }
+        }
+
+        [Fact]
+        public void VerifyResourceBuilder_WithAttributesEnVar()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(OtelEnvResourceDetector.EnvVarKey, "Key1=Val1,Key2=Val2");
+
+                InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider);
+
+                Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "Key1" && kvp.Value.ToString() == "Val1");
+                Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "Key2" && kvp.Value.ToString() == "Val2");
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(OtelEnvResourceDetector.EnvVarKey, null);
+            }
+        }
+
+        private static void InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider, Action<OpenTelemetryLoggerOptions> configure = null)
+        {
+            var exporter = new InMemoryExporter<LogRecord>(new List<LogRecord>());
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    configure?.Invoke(options);
+                    options.AddProcessor(new TestLogRecordProcessor(exporter));
+                });
+                builder.AddFilter(typeof(LoggerProviderTests).FullName, LogLevel.Trace);
+            });
+            var logger = loggerFactory.CreateLogger<LoggerProviderTests>();
+
+            provider = exporter.ParentProvider as OpenTelemetryLoggerProvider;
```

Comment:
How long would the provider live?

Reply:
I'm not certain.
After the method exists we don't need the provider, however we're holding a reference to one of it's members, `Resource`.